src/client.ts

```diff
@@ -145,9 +147,20 @@ export class JSONRPCClient<ClientParams = void>
     if (response.result !== undefined && !response.error) {
       return response.result;
     } else if (response.result === undefined && response.error) {
-      return Promise.reject(new Error(response.error.message));
+      return Promise.reject(
+        new JSONRPCErrorException(
+          response.error.code,
+          response.error.message,
+          response.error.data
+        )
+      );
     } else {
-      return Promise.reject(new Error("An unexpected error occurred"));
+      return Promise.reject(
+        new JSONRPCErrorException(
+          JSONRPCErrorCode.ParseError,
```

Comment:
Parse error is reserved for something else: https://www.jsonrpc.org/specification#error_object

> Invalid JSON was received by the server.
> An error occurred on the server while parsing the JSON text.

Also since this code hasn't yet reached the RPC pipeline, I think this should not be a JSON-RPC error. Can you please revert?

Reply:
You are quite right - reverted.